jaxrs-tck/src/main/java/jakarta/ws/rs/tck/common/JAXRSCommonClient.java

```diff
@@ -0,0 +1,1098 @@
+   *              if an error occurs
+   */
+  public void setup(String[] args, Properties p)   {
+    //TestUtil.logTrace("setup method JAXRSCommonClient");
+
+    String hostname = p.getProperty(SERVLETHOSTPROP);
+    String portnum = p.getProperty(SERVLETPORTPROP);
+    String tshome = p.getProperty(TSHOME);
+
+    assertTrue(!isNullOrEmpty(hostname),
+        "[JAXRSCommonClient] 'webServerHost' was not set in the build.properties.");
+    _hostname = hostname.trim();
+    assertTrue(!isNullOrEmpty(portnum),
+        "[JAXRSCommonClient] 'webServerPort' was not set in the build.properties.");
+    _port = Integer.parseInt(portnum.trim());
+    assertTrue(!isNullOrEmpty(tshome),
+        "[JAXRSCommonClient] 'tshome' was not set in the build.properties.");
+    _tsHome = tshome.trim();
+
+    //TestUtil.logMsg("[JAXRSCommonClient] Test setup OK");
+  }
+
+  /**
+   * <code>cleanup</code> is called by the test harness to cleanup after text
+   * execution
+   * 
+   * @exception Fault
+   *              if an error occurs
+   */
+  public void cleanup()   {
+    //TestUtil.logMsg("[JAXRSCommonClient] Test cleanup OK");
+  }
+
+  /*
+   * protected methods
+   * ========================================================================
+   */
+  /**
+   * <PRE>
+   * Invokes a test based on the properties
+   * stored in TEST_PROPS.  Once the test has completed,
+   * the properties in TEST_PROPS will be cleared.
+   * </PRE>
+   * 
+   * @ 
+   *           If an error occurs during the test run
+   */
+  protected void invoke()   {
+    //TestUtil.logTrace("[JAXRSCommonClient] invoke");
+    try {
+      _testCase = new WebTestCase();
+      setTestProperties(_testCase);
+      //TestUtil.logTrace("[JAXRSCommonClient] EXECUTING");
+      if (_useSavedState && _state != null) {
+        _testCase.getRequest().setState(_state);
+      }
+      if (_redirect != false) {
+        //TestUtil.logTrace("##########Call setFollowRedirects");
+        _testCase.getRequest().setFollowRedirects(_redirect);
+      }
+      _testCase.execute();
+      if (_saveState) {
+        _state = _testCase.getResponse().getState();
+      }
+    } catch (TestFailureException tfe) {
+      Throwable t = tfe.getRootCause();
+      if (t != null) {
+        //TestUtil.logErr("Root cause of Failure: " + t.getMessage(), t);
+      }
+      //throw new Fault("[JAXRSCommonClient] " + _testName
+      //    + " failed!  Check output for cause of failure.", tfe);
+    } finally {
+      _useSavedState = false;
+      _saveState = false;
+      _redirect = false;
+      clearTestProperties();
+    }
+  }
+
+  /**
+   * <PRE>
+   * Sets the appropriate test properties based
+   * on the values stored in TEST_PROPS
+   * </PRE>
+   */
+  protected void setTestProperties(WebTestCase testCase) {
+    //TestUtil.logTrace("[JAXRSCommonClient] setTestProperties");
+
+    //temporarily hardcode hostname & portnumber
+    String hostname = "localhost";
+    String portnum = "8080";
```

Comment:
```suggestion
    String portnum = System.getProperty("http.port", "8080");
```

This will allow vendors to use a port other than 8080 by setting the system property.  For example, Open Liberty uses 9080 by default.

Reply:
Indeed, while it sounds weird, I am not aware of any Jakarta specification which actually mandates that an implementation has customizable IP port (while I assume, *all* server have this). So Andy is right, this *must* be configurable on the TCK side.

Reply:
setting the property via jersey-tck/pom.xml